ansible_bender/callback_plugins/snapshoter.py

```diff
@@ -69,17 +72,50 @@ def _snapshot(self, task_result):
 
     @staticmethod
     def get_task_content(task: Task):
+        sha512 = hashlib.sha512()
         serialized_data = task.get_ds()
         if not serialized_data:
             # ansible 2.8
             serialized_data = task.dump_attrs()
         if not serialized_data:
             logger.error("unable to obtain task content from ansible: caching will not work")
             return
-        c = json.dumps(serialized_data, sort_keys=True).encode("utf-8")
+        c = json.dumps(serialized_data, sort_keys=True)
+
         logger.debug("content = %s", c)
-        m = hashlib.sha512(c)
-        return m.hexdigest()
+        sha512.update(c.encode("utf-8"))
+
+        # If task is a file action, cache the src.
+        # Take the contents of the src (could be file or directory)
+        # and concatinate it with the task config (or serialized_data).
+        #
+        # Pseudocode: hash(serialized_data + src_file)
+        #
+        # Note: serialized_data was grabbed above.
+        try:
+            task_config = json.loads(c)[task.action]
+            src_regex = "(?<=src\=).*?(?=\ )"
+            src = re.search(src_regex, task_config).group(0)
+            src_path = task.get_search_path()[0] + "/files/" + src
```

Comment:
this is pretty naive and would only work for roles

Reply:
This works for non-roles. Though I'm still not entirely happy. I wish there was a more elegant way to get the full path of a given source file. Like `task.get_file_search_path()` or something.
```
            src = task_config['args']['src']
            src_path = os.path.join(task.get_search_path()[0], "files", src)

            if(not(os.path.exists(src_path))):
                src_path = os.path.join(task.get_search_path()[0], src)
```